src/Symfony/Component/BrowserKit/AbstractBrowser.php

```diff
@@ -166,16 +166,42 @@ public function xmlHttpRequest(string $method, string $uri, array $parameters =
      */
     public function jsonRequest(string $method, string $uri, array $parameters = [], array $server = [], bool $changeHistory = true): Crawler
     {
-        $content = json_encode($parameters);
+        // Based on https://github.com/symfony/symfony/blob/v5.2.0/src/Symfony/Component/HttpFoundation/Request.php#L388-L404
+        // the logic in symfony/http-foundation we convert parameters to json content.
+        switch (strtoupper($method)) {
+            case 'POST':
+            case 'PUT':
+            case 'DELETE':
+            case 'PATCH':
+                $content = json_encode($parameters, \defined('JSON_THROW_ON_ERROR') ? \JSON_THROW_ON_ERROR : 0);
```

Comment:
We should not use `JSON_THROW_ON_ERROR` conditionnally here: the behavior must be the same whatever the version of php.

Reply:
Thx for feedback I removed the `JSON_THROW_ON_ERROR` to be consistent between all PHP versions.